hack/determine_install_upgrade_version_test.py

```diff
@@ -0,0 +1,41 @@
+import unittest
+import rpmUtils.miscutils as rpmutils
+from determine_install_upgrade_version import *
+
+class TestPackage(object):
+    def __init__(self, version, release, vra):
+        self.version = version
+        self.release = release
+        self.vra = vra
+
+class DetermineInstallUpgradeVersionTestCase(unittest.TestCase):
+	"Test for `determine_install_upgrade_version.py`"
+	PKG_NAME = "origin"
+	MATCHING_1_5 = ["1.5.0-0.4.el7.x86_64"]
+	MATCHING_1_4 = ["1.4.1-1.el7.x86_64"]
+	MATCHING_1_3 = ["1.3.1-1.el7.x86_64", "1.3.2-3.el7.x86_64"]
+	MATCHING_1_2 = ["1.2.0-0.el7.x86_64", "1.2.5-0.el7.x86_64"]
+
+	def test_get_matching_versions(self):
+		self.assertEqual(get_matching_versions(self.PKG_NAME, test_pkgs_objs, "1.5"), self.MATCHING_1_5)
+		self.assertEqual(get_matching_versions(self.PKG_NAME, test_pkgs_objs, "1.4"), self.MATCHING_1_4)
+		self.assertEqual(get_matching_versions(self.PKG_NAME, test_pkgs_objs, "1.3"), self.MATCHING_1_3)
+		self.assertEqual(get_matching_versions(self.PKG_NAME, test_pkgs_objs, "1.2"), self.MATCHING_1_2)
+
+
+if __name__ == '__main__':
+	test_pkgs = ["origin-1.2.0-0.el7.x86_64",
```

Comment:
Always prefer to place test data as a literal object in the test scope. Otherwise, you start giving tests more than one reason to fail, and that makes it harder to debug failures.